packages/logger/README.md

```diff
@@ -1,4 +1,320 @@
+        }
+      }
+    }
+  }
+}
+```
+@api is a [class decorator](https://www.typescriptlang.org/docs/handbook/decorators.html).  The class decorator,
+`api` is a function that is called when the class, HelloWorldController is instantiated.  In this case, `api` registers
+the API specification to the application.  The decorator site describes some relatively complex concept, but we can just read the class decorator section to get a high-level understanding of decorator.  The concept is the same for other types of decorators.  That's enough.
+
+In the above sample code, @inject is a property decorator defined by LoopBack Next.  It's a very important LoopBack Next concept.  It works this way: @inject acquires the value bound to the key: `defaultName` in the application's context, then assigns the value to `name` property when the class `HelloWorldControlle` is instantiated.
+
+Please note that we introduced another new term, `Context`.  Context is simply a memory space where LoopBack Next maintains key-value pairs.  There are two types of context: application context and request context.  In this blog, we deal with only application context.
+
+The LoopBack Next specific decorator, @inject can be used as a parameter decorator e.g.:
+```
+  helloWorld(@inject('defaultName') name: string) { ... }
+```
+In this case, the argument `name` is replaced with the value bound to the key `defaultName`.  @inject acquires the value bound to the `defaultName` key in the application's context, then assigns the value to `name` argument when the helloWorld method is invoked.
+
+Who binds `defaultName` key to the value?  It depends.  You can do that by app.bind(`defaultName`).to('Ted Johnson').  You can `get` the value string by app.getSync('defaultName').
+
+More important usage of @inject is discussed in `Component and Provider` section.
+
+### Components and Providers
+
+Component defines a functionality.  LoopBack Next application can be viewed as a collection of components.  In a component, one or more providers implement the functionality.  In the Logger provider example below, (a) 'logger' key is  bound to the provider instance when Application is instantiated, and (b) the client is acquiring the logger instance by app.getSync('logger') and using the logger to display the message, 'My application has started.'.
+
+Client:
+```
+const app = new Application({
+  components: [AuthenticationComponent, LoggerComponent],
+});
+
+const logger = app.getSync('logger');
+logger.info('My application has started.');
+
+```
+
+Implementor:
+```
+import {Component} from '@loopback/core';
+const key = 'logger';
+
+export class LoggerComponent implements Component {
+  providers = {
+    [key]: LoggerProvider,  // [key] is a computed property name
+  }
+}
+```
+  Note: What's computed property name ?  See http://es6-features.org/#ComputedPropertyNames
+```
+import {Provider} from '@loopback/context';
+
+export class LoggerProvider implements Provider<SimpleLogger> {
```

Comment:
Let's use `Logger` instead of `SimpleLogger`. Using "Simple" as a name prefix is a sort of an anti-pattern: "simple" does not tell us any new information about the class behavior and how does it differ from other "complex?" loggers.